x-pack/plugins/cases/server/client/metrics/get_case_metrics.ts

```diff
@@ -67,31 +64,45 @@ const buildHandlers = (
   params: CaseMetricsParams,
   casesClient: CasesClient,
   clientArgs: CasesClientArgs
-): Map<string, MetricsHandler> => {
-  const handlers = [
+): Set<MetricsHandler> => {
+  const handlers: MetricsHandler[] = [
     new Lifespan(params.caseId, casesClient),
     new AlertsCount(params.caseId, casesClient, clientArgs),
-    new AlertDetails(),
+    new AlertDetails(params.caseId, casesClient, clientArgs),
     new Connectors(),
   ];
 
-  const handlersByFeature = new Map<string, MetricsHandler>();
+  const uniqueFeatures = params.features.reduce((acc, feature) => {
+    acc.add(feature);
+    return acc;
+  }, new Set<string>());
+
+  const handlerFeatures = new Set<string>();
+  const handlersToExecute = new Set<MetricsHandler>();
   for (const handler of handlers) {
-    // assign each feature to the handler that owns that feature
-    handler.getFeatures().forEach((value) => handlersByFeature.set(value, handler));
+    for (const handlerFeature of handler.getFeatures()) {
+      if (uniqueFeatures.has(handlerFeature)) {
+        handler.setupFeature?.(handlerFeature);
+        handlersToExecute.add(handler);
+      }
+
+      handlerFeatures.add(handlerFeature);
+    }
   }
 
-  return handlersByFeature;
+  checkAndThrowIfInvalidFeatures(params, handlerFeatures);
+
+  return handlersToExecute;
```

Comment:
We now only return the handles that need to be executed, instead of a map of all of them. That way we can avoid looping through the features and accidentally calling the alerts handler twice (once for users and once for hosts).